diff --git a/src/imp_std.rs b/src/imp_std.rs
index d7dda96..f461c3d 100644
--- a/src/imp_std.rs
+++ b/src/imp_std.rs
@@ -299,6 +299,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(not(target_os = "android"))]
     fn poison_bad() {
         static O: OnceCell<()> = OnceCell::new();
 
@@ -320,6 +321,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(not(target_os = "android"))]
     fn wait_for_force_to_finish() {
         static O: OnceCell<()> = OnceCell::new();
 
