diff --git a/devices/Android.bp b/devices/Android.bp
index a1ac9363..ae2fdde5 100644
--- a/devices/Android.bp
+++ b/devices/Android.bp
@@ -83,7 +83,15 @@ rust_test {
                 "libgpu_display",
                 "librutabaga_gfx",
             ],
-            shared_libs: ["libgfxstream_backend"],
+            shared_libs: ["libvirglrenderer", "libgfxstream_backend"],
+        },
+        android: {
+            shared_libs: ["libdrm"],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
         },
     },
 }
